src/dataIntegration/transcriptomics/MOOMIN/moomin.m

```diff
@@ -0,0 +1,376 @@
+%    enumerate:         integer to determine the maximum number of alternative optimal
+%                       solutions that are enumerated  (default 1 ie only one solution)
+%    precision:         integer to determine up to how many significant numbers the
+%                       weights are evaluated. Will influence the uniqueness of weight
+%                       values. (default 7)
+%    solverTimeLimit:	time limit for the MILP solver (default 1000)
+%    solverPrintLevel:  print level parameter for the MILP solver (default 0)
+%    solverParameters:	struct containing solver parameters to be passed on to
+%						solverCobraMILP
+%
+% OUTPUTS:
+%    model:             input model with additional fields containing outputs of the
+%                       algorithm. Rows correspond to reactions. The colours are coded by
+%                         2 - reverse red (r.red) ie increase, reaction in reverse
+%                         1	- red ie increase
+%                         0 - grey ie no change
+%                         -1 - blue ie decrease
+%                         -2 - reverse blue (r.blue) ie decrease, reaction in reverse
+%                         6	- yellow ie unspecified change in a reversible reaction
+%                       Additional fields are
+%
+%                         * .inputColours - colours inferred solely based on data
+%                         * .outputColours - matrix, colours inferred by the algorithm
+%                           columns correspond to alternative optimal solutions
+%                         * .weights - the reaction weights used in the algorithm
+%                         * .leadingGenes -  the genes "responsible" for the colour and
+%                           weight of a reaction
+%                         * .frequency - how often a reaction is coloured in a solution
+%                         * .combined - an attempted consensus between all optimal
+%                           solutions (colour differs between solutions -> 6)
+%                         * .PPDE - PPDEs of the model genes. -1 for missing values
+%                         * .FC - fold changes of the model genes. 0 for missing values
+%    MILPsolutions      raw outputs of 'solveCobraMILP'
+%    MILPproblem        the final MILP-problem solved (or was attempted to be solved)
+%
+% `Pusa et al. (2019). MOOMIN – Mathematical explOration of 'Omics data on a MetabolIc Network.`
+%
+% .. Author: - Taneli Pusa 01/2020
+
+	pThresh = 0.9;
+	alpha = 3;
+	beta = 2; % additional parameter used in the weight function
+	epsilon = 1;
+	useStoichiometry = 1;
+	enumerate = 1;
+	precision = 7;
+	
+	% suggested solver parameters, override by passing a parameter structure through
+	%	'solverParameters' option
+	tolerance = 1e-6;
+    solverParameters.intTol = tolerance;
+    solverParameters.absMipGapTol = tolerance;
+    solverParameters.feasTol = tolerance;
+    solverParameters.optTol = tolerance;
```

Comment:
Why don't you use the default COBRA parameters if no explicit parameters are given?
Setting the parameters in this function is (in my opinion) unexpected behavior which can easily lead to misinterpretation of its results.

Reply:
I change the parameters to improve performance. The MILP can become quite difficult to solve. But if this is unwanted, I can change it.

Reply:
Hmm...
Personally, I'd call this unexpected behaviour.
I would rather put in a "suggested parameters" information in the Documentation and/or a warning when using parameters which deviate from those you suggest. But changing them "silently" in a function (even if only for that function) is kind of odd and can potentially lead to unexpected problems which are then very hard to figure out.

Reply:
Duly noted, will change.